src/Confluent.SchemaRegistry.Serdes.Protobuf/ProtobufUtils.cs

```diff
@@ -300,40 +300,45 @@ private static ISet<string> GetInlineTags(FieldDescriptorProto fd)
 
         public static FileDescriptorSet Parse(string schema, IDictionary<string, string> imports)
         {
-            IDictionary<string, string> allImports = new Dictionary<string, string>(BuiltIns);
-            imports?.ToList().ForEach(x => allImports.Add(x.Key, x.Value));
-            
-            var fds = new FileDescriptorSet();
-            fds.FileSystem = new ProtobufImports(allImports);
-            
-            fds.Add("__root.proto", true, new StringReader(schema));
-            foreach (KeyValuePair<string, string> import in allImports)
+            var allImports = imports != null
+                ? BuiltIns
+                    .Concat(imports)
+                    .GroupBy(s => s.Key) // remove duplicates, prefer built ins
+                    .ToDictionary(s => s.Key, s => s.First().Value)
+                : BuiltIns;
+            var fds = new FileDescriptorSet
             {
-                fds.AddImportPath(import.Key);
-                
-            }
+                FileSystem = new ProtobufImports(allImports)
+            };
+            fds.Add("__root.proto", true, new StringReader(schema));
+            fds.AddImportPath("/"); // all imports are from the root of the ProtobufImports file system
```

Comment:
This was the actual problem. The FileDescriptorSet resolves the imports according to the import path+the name of the file so with the original code, it was never finding the actual imports.